etcdserver/v3demo_server.go

```diff
@@ -398,19 +422,12 @@ func applyTxn(kv dstorage.KV, le lease.Lessor, rt *pb.TxnRequest) (*pb.TxnRespon
 		}
 	}
 
-	// TODO: check potential errors before actually applying anything
-
-	var reqs []*pb.RequestUnion
 	if ok {
 		reqs = rt.Success
 	} else {
 		reqs = rt.Failure
 	}
 
-	if err := checkRequestLeases(le, reqs); err != nil {
```

Comment:
Old behavior was etcd would permit txns with bad requests if those requests won't run (my thinking was it'd be a little cheaper to wait until knowing which reqs were necessary). I'm not sure whether checking everything up front is better, aside from avoiding an empty txn on the (hopefully unlikely) failure path.

Anyway, the one or two tests that expect the permissive version will need updating if you're going with the up-front checks.